dash-spv/src/storage/disk/state.rs

```diff
@@ -434,6 +434,9 @@ impl DiskStorageManager {
         // Save all dirty segments
         super::segments::save_dirty_segments(self).await?;
 
+        // Ensure the reverse index reflects every stored header before we exit
+        self.flush_header_index().await?;
+
         // Shutdown background worker
         if let Some(tx) = self.worker_tx.take() {
             let _ = tx.send(super::manager::WorkerCommand::Shutdown).await;
```

Comment:
```suggestion
            let index = self.header_hash_index.read().await.clone();
            let _ = tx.send(super::manager::WorkerCommand::SaveIndex { index }).await;
            let _ = tx.send(super::manager::WorkerCommand::Shutdown).await;
```

You can probably just do this? (Didn't try though, just figured while reviewing)

- Reuse the existing code to write the index
- Lets the write happen in the background process.

Reply:
Yes, thanks